ProfileManager.js

```diff
@@ -115,6 +122,70 @@ export default class ProfileManager {
     return await this._getAgentContent({profileAgentRecord});
   }
 
+  /**
+   * Gets a capability specified by its reference ID for profile agent from
+   * a cache or delegates one on the fly if it does not exist.
+   *
+   * @param {object} options - The options to use.
+   * @param {string} options.id - The reference ID of the zcap.
+   * @param {string} options.profileAgent - The profile agent requesting a zcap.
+   * @param {boolean} options.useEphemeralSigner - Flag to enable invoking
+   *   capabilities with the ephemeral invocation signer associated with the
+   *   currently authenticated session, default `true`.
+   *
+   * @returns {Promise<object>} The capability for the profile agent.
+   */
+  async getAgentCapability({id, profileAgent, useEphemeralSigner = true}) {
+    const capabilityCacheKey = `profileAgent-${profileAgent.id}-zcaps`;
+    const capabilityKey = `${capabilityCacheKey}-${id}-${useEphemeralSigner}`;
+
+    if(!this._cacheContainer.has(capabilityCacheKey)) {
+      this._cacheContainer.set(capabilityCacheKey, new Cache());
+    }
+
+    const capabilityCache = this._cacheContainer.get(capabilityCacheKey);
+    const agentZcap = capabilityCache.get(capabilityKey);
+    if(agentZcap) {
+      return agentZcap;
+    }
+
+    const promise = (async () => {
+      let zcap;
+      const agentSigner = await this.getAgentSigner(
+        {profileAgentId: profileAgent.id, useEphemeralSigner: false});
+      const originalZcap = profileAgent.zcaps[id];
+      if(!originalZcap) {
+        const {id: profileAgentId} = profileAgent;
+        throw new Error(
+          `The agent "${profileAgentId}" does not have the zcap: "${id}"`);
+      }
+      if(useEphemeralSigner) {
```

Comment:
It would improve readability and reduce nesting to do `if(!useEphemeralSigner)` first.

Reply:
Could remove the extra `zcap` var as well.